### PR TITLE
[Issue 14013][broker]make offloadStatus method async

### PR DESCRIPTION
**Pull request by HQebupt**
Master Issue: #14013 
### Motivation

See #14013 

### Verifying this change

Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -3759,12 +3759,14 @@ protected void internalTriggerOffload(boolean authoritative, MessageIdImpl messa
...
-        PersistentTopic topic = (PersistentTopic) getTopicReference(topicName);
-        return topic.offloadStatus();
+    protected void internalOffloadStatus(AsyncResponse asyncResponse, boolean authoritative) {
+        validateTopicOwnershipAsync(topicName, authoritative)
+                .thenCompose(__ -> validateTopicOperationAsync(topicName, TopicOperation.OFFLOAD))
+                .thenCompose(__ -> getTopicReferenceAsync(topicName))
+                .thenAccept(topic -> {
+                    OffloadProcessStatus offloadProcessStatus = ((PersistentTopic) topic).offloadStatus();
+                    asyncResponse.resume(offloadProcessStatus);
+                });
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3769]: Missed exception handling here?

Comment (HQebupt) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:3783]: Thanks, PTAL again. @codelipenghui

**Review by mattisonchao — APPROVED** [review]
LGTM +1.
but I have two suggestions:

Remove asyncResponse from the internal method, and unified exception and result handling are performed at the controller layer, which makes the internal method purer.
Have you checked to see if there is a test for the modified method ?

**Comment by HQebupt** [comment]
> LGTM +1. but I have two suggestions:
> 
> Remove asyncResponse from the internal method, and unified exception and result handling are performed at the controller layer, which makes the internal method purer. Have you checked to see if there is a test for the modified method ?

@mattisonchao Thank you. I got your meaning. I will try your way in the following refactor. Truly, there are test failure due to the new modification in my other pull requests.

**Comment by HQebupt** [comment]
@codelipenghui @Jason918 PTAL, thanks

**Review by Jason918 — APPROVED** [review]

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**